src/App.js

```diff
@@ -14,11 +16,119 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  clearSelected = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  renderTitle = (array) => {
+    switch (array.length) {
+      case 0:
+        return (
+          'No goods selected'
+        );
+
+      case 1:
+        return (
+          `${array[0]} is selected`
+        );
+
+      default:
+        return (
+          `${array.slice(0, -1).join(', ')} 
+          and ${array[array.length - 1]} 
+          are selected`
+        );
+    }
+  };
+
+  addProduct(product) {
+    this.setState((prevState) => {
+      prevState.selectedGoods.push(product);
+
+      return {
+        selectedGoods: prevState.selectedGoods,
+      };
+    });
```

Comment:
we don't change state, we create a new array in setState

```suggestion
    this.setState((prevState) => ({
        selectedGoods: [...prevState, product],
    }));
```